### ISSUE TITLE
Possible bug with sprite:set_frame_delay(frame_delay) not updated

### ISSUE THREAD

**Issue by Diarandor** [christopho/solarus#856]
If we start an animation on a sprite and then change the frame delay with:
```
sprite:set_frame_delay(frame_delay)
```
the frame delay is not applied to the current animation. It is necessary to restart the animation to have the new chosen frame delay.

I do not know if this is intended or a bug.
In case this was intended, then the API should tell that we need to restart the animation.

**Comment by christopho** [comment]
I just checked and it works correctly. The frame delay is applied to the current animation. Can you provide more details to reproduce the bug?

**Comment by Diarandor** [comment]
New info:
I found this bug when I used a custom entity as a shadow for the custom jump of the hero, so this issue happens at least for sprites of custom entities. The shadow has an animation (it becomes smaller and later it recovers the original size). 

In my feather (or jumping) script I start the animation and then I immediately change the frame delay. (The frame delay is changed depending on the duration of the jump, which is chosen in the script.) I have noticed that modifying the frame delay of the animation directly from the Editor affects to the time between frames during the jump. But my code (when I modify the frame delay in the script) also changes the time between frames. So it's something like both things are affecting to the time between frames, which is quite strange (it should not depend on the original frame delay since I am changing the value in the script).

I hope this info helps to detect this weird problem.

**Comment by christopho** [comment]
Here is the expected behavior:
Changing the frame delay from code temporarily replaces the frame delay of the current animation. When another animation becomes the current one, the frame delay from its the sprite file is used again.
Do you observe something different than that?

**Comment by Diarandor** [comment]
I am starting to think that the problem is that the engine wailts for the current frame to end, before changing to the new frame delay. I have tested and I think that is the problem. (The frame delay seems to be applied correctly, so there would be no such a bug as the one I posted initially.)

**Comment by christopho** [comment]
Indeed, the frame delay is applied after the next frame. So this can still be considered as a bug because there is no way to change the duration of the very first frame.
